packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -373,7 +375,13 @@ abstract class ScrollableState<T extends Scrollable> extends State<T> {
     });
     PageStorage.of(context)?.writeState(context, _scrollOffset);
     new ScrollNotification(this, _scrollOffset).dispatch(context);
+    final needsScrollStart = !_isBetweenOnScrollStartAndOnScrollEnd;
+    if (needsScrollStart)
+      dispatchOnScrollStart();
     dispatchOnScroll();
```

Comment:
either add the comment `// sets _isBetweenOnScrollStartAndOnScrollEnd` on the line that calls `dispatchOnScrollStart()`, or, add the `assert(_isBetweenOnScrollStartAndOnScrollEnd)` assert before the call to `dispatchOnScroll()` too, so it's clearer that it's getting set there. Otherwise, it's not obvious whether you're expecting that to be set to true by scroll start or scroll.